src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -230,4 +230,27 @@ public function loadRoutes(LoaderInterface $loader): RouteCollection
 
         return $collection;
     }
+
+    /**
+     * Returns the kernel parameters.
+     *
+     * @return array<string, array|bool|string|int|float|\UnitEnum|null>
+     */
+    protected function getKernelParameters(): array
+    {
+        $parameters = parent::getKernelParameters();
+        $bundlesPath = $this->getBundlesPath();
+        $allBundles = !is_file($bundlesPath) ? [FrameworkBundle::class => ['all' => true]] : require $bundlesPath;
+        $knownEnvs = [$this->environment => true];
+
+        foreach ($allBundles as $envs) {
+            $knownEnvs += $envs;
+        }
+        unset($knownEnvs['all']);
+        $parameters['.kernel.config_dir'] = $this->getConfigDir();
+        $parameters['.kernel.known_envs'] = $knownEnvs;
```

Comment:
shouldn't we use `array_keys($knownEnvs)` here, to produce a parameter containing a list of envs instead of having them as keys ? PhpConfigReferenceDumpPass is currently doing it anyway, but we might reuse it in other passes that would need the same.